clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -519,12 +518,10 @@ public void subscribe(final Collection<String> topics) {
                 }
 
                 log.info("Subscribed to topic(s): {}", String.join(", ", topics));
-                if (subscriptions.subscribeToShareGroup(new HashSet<>(topics)))
-                    metadata.requestUpdateForNewTopics();
 
                 // Trigger subscribe event to effectively join the group if not already part of it,
                 // or just send the new subscription to the broker.
-                applicationEventHandler.add(new ShareSubscriptionChangeEvent());
+                applicationEventHandler.addAndGet(new ShareSubscriptionChangeEvent(topics));
```

Comment:
nit: Would it makes sense to move the info message above to follow this? Either that or change to "Subscribing" (such as "Unsubscribing" below), but I think it is more useful to print the message after the operation has completed.

Reply:
I agree, and suggest we change it consistently in the other 2 consumers too, all have the exact same message with "inverted" order of log/action.